src/BaseStrategy.sol

```diff
@@ -412,36 +434,38 @@ abstract contract BaseTokenizedStrategy {
      *
      * @param _amount The amount of asset to attempt to free.
      */
-    function shutdownWithdraw(uint256 _amount) external onlySelf {
+    function shutdownWithdraw(uint256 _amount) external virtual onlySelf {
         _emergencyWithdraw(_amount);
     }
 
     /**
-     * @dev Funciton used on initialization to delegate call the
-     * TokenizedStrategy to setup the default storage for the strategy.
+     * @dev Funciton used to delegate call the TokenizedStrategy with
+     * certain `_calldata` and return any return values.
      *
-     * We cannot use the `TokenizedStrategy` variable call since this
-     * contract is not deployed fully yet. So we need to manually
-     * delegateCall the TokenizedStrategy.
+     * This is used to setup the intial storage of the strategy, and
+     * can be used by strategist to forward any other call to the
+     * TokenizedStrategy implementation.
      *
-     * This is the only time an internal delegateCall should not
-     * be for a view function
+     * @param _calldata The abi encoded calldata to use in delegatecall.
+     * @return . The return value if the call was successful in bytes.
      */
-    function _init(
-        address _asset,
-        string memory _name,
-        address _management,
-        address _performanceFeeRecipient,
-        address _keeper
-    ) private {
-        (bool success, ) = tokenizedStrategyAddress.delegatecall(
-            abi.encodeCall(
-                ITokenizedStrategy.init,
-                (_asset, _name, _management, _performanceFeeRecipient, _keeper)
-            )
-        );
+    function _delegate(bytes memory _calldata) internal returns (bytes memory) {
```

Comment:
I'm not 100% sure i like the name of this function, but I could live with it.

Reply:
https://github.com/yearn/tokenized-strategy/pull/67/commits/7934b03d794607dc2e2e33c0cebea672d84ed615